namespaces/mson-namespace.md

```diff
@@ -0,0 +1,497 @@
+# MSON Namespace
+
+This document extends [Refract][] Specification with new element types necessary to build [MSON][] DOM.
+
+# Content
+
+This namespace defines following elements:
+
+1. [MSON Element](#mson-element-element)
+1. [Boolean Type](#boolean-type-boolean-element)
+1. [String Type](#string-type-string-element)
+1. [Number Type](#number-type-number-element)
+1. [Array Type](#array-type-array-element)
+1. [Object Type](#object-type-object-element)
+1. [Property Type](#property-type-property-element)
+1. [Enum Type](#enum-type-mson-element)
+
+# About this Document
+
+This document conforms to RFC 2119, which says:
+
+> The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+[MSON][] is used throughout this document.
+
+## Expanded Element
+
+MSON is built around the idea of defining recursive data structures. To provide abstraction, for convenience reasons and to not repeat ourselves, these structures can be named (using an _identifier_) and reused. In [MSON][], the reusable data structures are called _Named Types_.
+
+Often, before an MSON DOM can be processed, referenced _Named Types_ have to be resolved. Resolving references to _Named Types_ is tedious and error prone. As such an MSON processor can resolve references to produce a complete MSON DOM. That is, a DOM that does not include unresolved references to other data structures. This is referred to as _reference expansion_ or simply _expansion_.
+
+In other words, an expanded element is one that does not contain any _Identifier_ (defined bellow) referencing any other elements than those defined in JSON or MSON namespaces.
```

Comment:
sp below. Also, did you mean to leave in the JSON namespace. Shouldn't this just use the refract namespace for primitives. Maybe I am misunderstanding.